Description: Explicitly link to all required libraries for binutils-gold.
Author: Stefano Rivera <stefano@rivera.za.net>
Last-Update: 2010-11-08
Index: elmerfem/front/configure.in
===================================================================
--- elmerfem.orig/front/configure.in
+++ elmerfem/front/configure.in
@@ -45,6 +45,7 @@
 
 AC_CHECK_LIB(stdc++, main)
 AC_CHECK_LIB(dl, main)
+AC_CHECK_LIB(X11, XGetErrorText)
 
 LIBS="$LIBS $MATC_LIBS $EIOC_LIBS $LIBS $GL_LIBS $GLU_LIBS $TCLTK_LIBS"
 ACX_PLATFORM_DEFS
Index: elmerfem/post/configure.in
===================================================================
--- elmerfem.orig/post/configure.in
+++ elmerfem/post/configure.in
@@ -50,6 +50,7 @@
 sinclude(acx_gl.m4)
 ACX_CHECK_GL
 ACX_CHECK_GLU
+AC_CHECK_LIB(X11, XGetErrorText)
 ACX_TCLTK([],[AC_MSG_ERROR([Tcl/tk not found, use TCLTK_LIBS environment variable or --with-tcltk="-L/path/to/ -ltcl8.4 -ltk8.4" to point to it's location.])])
 CPPFLAGS="$CPPFLAGS $TCLTK_INCLUDE"
 
Index: elmerfem/ElmerGUI/ElmerGUI.pri
===================================================================
--- elmerfem.orig/ElmerGUI/ElmerGUI.pri
+++ elmerfem/ElmerGUI/ElmerGUI.pri
@@ -77,9 +77,21 @@
 unix {
    VTK_INCLUDEPATH = /usr/include/vtk-5.4
    VTK_LIBPATH = /usr/lib
-   VTK_LIBS = -lvtkHybrid \
-              -lvtkWidgets \
-    	      -lQVTK
+   VTK_LIBS = -lQVTK \
+              -lvtkCommon \
+              -lvtkDICOMParser \
+              -lvtkFiltering \
+              -lvtkGenericFiltering \
+              -lvtkGraphics \
+              -lvtkHybrid \
+              -lvtkIO \
+              -lvtkImaging \
+              -lvtkInfovis \
+              -lvtkNetCDF \
+              -lvtkRendering \
+              -lvtkViews \
+              -lvtkVolumeRendering \
+              -lvtkWidgets
 }
 
 win32 {
@@ -129,7 +141,23 @@
 unix {
    OCC_INCLUDEPATH = /usr/include/opencascade
    OCC_LIBPATH = /usr/lib
-   OCC_LIBS = -lTKBRep -lTKSTL -lTKSTEP -lTKIGES
+   OCC_LIBS = -lTKSTL \
+              -lTKBRep \
+              -lTKernel \
+              -lTKG2d \
+              -lTKG3d \
+              -lTKGeomAlgo \
+              -lTKGeomBase \
+              -lTKMath \
+              -lTKMesh \
+              -lTKShHealing \
+              -lTKSTEP \
+              -lTKSTEP209 \
+              -lTKSTEPAttr \
+              -lTKSTEPBase \
+              -lTKIGES \
+              -lTKTopAlgo \
+              -lTKXSBase
 }
 
 win32 {
